src/libstd/time.rs

```diff
@@ -521,6 +530,26 @@ mod tests {
 
         let second = Duration::new(1, 0);
         assert_almost_eq!(a - second + second, a);
+
+        // checked_add_duration will not panic on overflow
+        let mut maybe_t = Some(Instant::now());
+        let max_duration = Duration::from_secs(u64::max_value());
+        // in case `Instant` can store `>= now + max_duration`.
+        for _ in 0..2 {
+            maybe_t = maybe_t.and_then(|t| t.checked_add(max_duration));
+        }
+        assert_eq!(maybe_t, None);
+
+        // checked_add_duration calculates the right time and will work for another year
+        let year = Duration::from_secs(60 * 60 * 24 * 365);
+        assert_eq!(a + year, a.checked_add(year).unwrap());
+
+        // Check that the checked_add_duration implementations seem approximately correct
+        let start = Instant::now();
+        thread::sleep(Duration::from_millis(500));
```

Comment:
I'm aware that time based tests are incredibly unreliable and usually not a good thing. But I felt that given the fact that the different implementations use time in vastly different scales it would be fairly easy to accidentally treat nanos as millis or similar. As such I thought something like this might help catch errors on the magnitude scale.

Reply:
I wonder if perhaps you could do `let end = start + Duration::from_millis(500)`? I'm not sure that it's necessary to mix-in `Instant::now()` testing here perhaps?

Reply:
Then the sleep is of no use. The sleep + `Instant::now()` is the way to create two instants separated in time without using the add functionality of `Instant`. Or am I missing something?

The `a +year` test just a few lines above make sure the normal add and the checked add produce the same results.

Reply:
Ah ok I think I see what you want to test, in that case I think it's fine to omit this test itself. It's true that `Instant::now` is notoriously hard to test correctly, so it's ok to just avoid it  on this one!

Reply:
I removed that part of the test completely :+1: